application/ui/components/input/custom-object.jsx

```diff
@@ -0,0 +1,206 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+var ObjectParameter;
+
+module.exports = ObjectParameter = React.createClass({
+
+    displayName : 'ObjectParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values : []
+        };
+    },
+
+    getValue : function()
+    {
+        return this.state.values;
+    },
+
+    getDefaultValue : function()
+    {
+        return {
+            key: '',
+            value: '',
+            inputType: 'string'
+        };
+    },
+
+    getInputs : function()
+    {
+        var inputs   = [],
+            instance = this;
```

Comment:
Not that this makes a big difference here, but FYI we've been using `component = this` as the name.